### PR TITLE
Transposed convolution (deconvolution)

### PR DESCRIPTION
**Pull request by vloncar**
This PR adds `QConv2DTranspose` which is useful in autoencoders. I can also add `QConv1DTranspose` but the equivalent Keras layer is only avaiable in nightly TF releases, not in stable channel yet.

### CODE REVIEW

File: examples/example_mnist_ae.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+RESHAPED = 784
+
+x_train = x_train.astype("float32")
+x_test = x_test.astype("float32")
+
+x_train = x_train[..., np.newaxis]
+x_test = x_test[..., np.newaxis]
+
+x_train /= 255.0
```
Comment (zhuangh) [inline examples/example_mnist_ae.py:64]: shall we use 256 since it is powers-of-two number.

**Comment by vloncar** [comment]
@zhuangh I'll make the changes you requested over the weekend. I didn't forget, I was just working on something else.

**Comment by zhuangh** [comment]
@vloncar awesome. thanks!

File: examples/example_mnist_ae.py (hunk 1 above)
Comment (vloncar) [reply examples/example_mnist_ae.py:64]: This code is based on the other mnist examples and they all use 255. It should be the same, no?

**Comment by vloncar** [comment]
@zhuangh do you have time to review this? thanks!

File: examples/example_mnist_ae.py (hunk 1 above)
Comment (zhuangh) [reply examples/example_mnist_ae.py:64]: My thought: 256 is power of two and more hardware friendly for the implementation.

**Comment by zhuangh** [comment]
Thanks @vloncar . And pull into the internal system for other round of review.

**Review by zhuangh — APPROVED** [review]

**Comment by zhuangh** [comment]
ping @lishanok

**Comment by zhuangh** [comment]
Hi @vloncar could you do me a favor to resolve the conflicting file qkeras/utils.py. we want to merge this PR soon. sorry for the delay.

cc @lishanok

File: qkeras/utils.py [hunk 2]
```diff
@@ -761,7 +764,7 @@ def quantized_model_debug(model, X_test, plot=False):
     if alpha != 1.0:
       print(" a[{: 8.4f} {:8.4f}]".format(np.min(alpha), np.max(alpha)))
     if plot and layer.__class__.__name__ in [
-      "QConv1D", "QConv2D", "QDense", "QActivation", 
+      "QConv1D", "QConv2D", "QConv2DTranspose" "QDense", "QActivation",
```
Comment (lishanok) [inline qkeras/utils.py:741]: Add comma here

**Comment by lishanok** [comment]
Hi @vloncar could you also sync qkeras/autoqkeras_internal.py with the latest code? We just did some changes to it. Thanks!

**Comment by vloncar** [comment]
@lishanok I don't see any conflicts with the latest change (Activation and limits).

**Comment by zhuangh** [comment]
I modified using https://github.com/google/qkeras/commit/e3d677867707e380f4d1292386883b6232d7d55b  to sync with https://github.com/google/qkeras/commit/e3d677867707e380f4d1292386883b6232d7d55b

**Pull request merged**